app/controllers/roles_controller.rb

```diff
@@ -46,8 +46,33 @@ def direct_memberships
     render json: json
   end
   
+  # Find all members of this role.
+  #
+  # For each member, return the full details of the grant as a
+  # JSON object
+  #
+  # +params[:count] returns only the number of members
+  # +params[:limit] and [:offset] control the paging
+  # +params[:search] returns only the members that match the search string
+  def members
+    filter_options, render_options = members_options
+
+    render_options[:order_by] ||= :member_id
+
+    members = @role.memberships_dataset
```

Comment:
Make a method for this too.... it's really nice if the controllers are just a sequence of high-level instructions, without implementation detail